io/pci/EEH.py

```diff
@@ -261,3 +285,31 @@ def is_baremetal():
         if 'PowerNV' in genio.read_file("/proc/cpuinfo").strip():
             return True
         return False
+
+    def get_paths(self, fc_host):
+        '''
+        returns the list of paths coressponding to the given fc_host
+        '''
+        paths = []
+        split_pattern = r':|\.'
+        fc_host = ':'.join(re.split(split_pattern, fc_host)[0:3])
```

Comment:
are you trying to split 
ls -l /sys/block/ | grep -i host0
lrwxrwxrwx. 1 root root 0 Nov 17 00:17 sda -> ../devices/vio/300000da/host0/rport-0:0-0/target0:0:0/0:0:0:0/block/sda
lrwxrwxrwx. 1 root root 0 Nov 17 00:17 sdb -> ../devices/vio/300000da/host0/rport-0:0-1/target0:0:1/0:0:1:0/block/sdb

just wanted to understand why this lines 
        split_pattern = r':|\.'
        fc_host = ':'.join(re.split(split_pattern, fc_host)[0:3])

when you already passing fc_host  as host0

Reply:
@abdhaleegit This split I performed on PCI Address(Eg: 001a:01:00.0).
In input file we are passing only one PCI address. When a PCI device reset happens, all the function device will also get reset. So to capture information from other function, I did split on '001a:01:00.0' and forming a new string '001a:01:00', It's used to capture all the disks which are coming from this PCI device.